massa-execution-worker/src/speculative_deferred_calls.rs

```diff
@@ -0,0 +1,517 @@
+//! Speculative async call registry.
+
+use crate::active_history::ActiveHistory;
+use massa_deferred_calls::{
+    registry_changes::DeferredRegistryChanges, DeferredCall, DeferredSlotCalls,
+};
+use massa_execution_exports::ExecutionError;
+use massa_final_state::FinalStateController;
+use massa_models::{
+    address::Address,
+    amount::Amount,
+    config::{
+        DEFERRED_CALL_BASE_FEE_MAX_CHANGE_DENOMINATOR, DEFERRED_CALL_MAX_FUTURE_SLOTS,
+        DEFERRED_CALL_MIN_GAS_COST, DEFERRED_CALL_MIN_GAS_INCREMENT, MAX_ASYNC_GAS,
+    },
+    deferred_call_id::DeferredCallId,
+    slot::Slot,
+};
+use parking_lot::RwLock;
+use std::{cmp::max, sync::Arc};
+
+const TARGET_BOOKING: u128 = (MAX_ASYNC_GAS / 2) as u128;
```

Comment:
Does this need calibration? Why is it this value?

Reply:
all constants in the math will need a theoretical calibration